### PR TITLE
Archive project

### PR DESCRIPTION
**Pull request by gabeholley222**
Able to Archive and Restore projects

### CODE REVIEW

**Comment by gabeholley222** [comment]
Messed up and don't actually shows changed files

File: src/ypd/templates/project.html [hunk 1]
```diff
@@ -28,6 +28,16 @@ <h1>{{project.title}}</h1>
...
+    {% if project.can_be_modified_by(user) %}
+        <div class="projectButton">
+            <a href="{{url_for('ProjectView:archive', id=project.id, is_provided=(project is provided))}}">
+                <button>
+                    🗑️
+                </button>
+            </a>
+        </div>
+    {% endif %}
+
```
Comment (caydenreynolds) [inline src/ypd/templates/project.html:40]: How can you tell if a project has been archived just by looking at it?

File: src/ypd/servlets/project_servlet.py [hunk 2]
```diff
@@ -59,6 +59,17 @@ def favorite(self, project):
         return redirect(url_for('ProjectView:view', id=project.id,
                                 is_provided=Tests.is_provided_test(project)))
 
+    # pull data from HTML form
+    @route('/archive', methods=('GET', 'POST'))
```
Comment (caydenreynolds) [inline src/ypd/servlets/project_servlet.py:63]: Why accept POST? You aren't using a form for this

File: src/ypd/model/project.py [hunk 3]
```diff
@@ -105,6 +105,15 @@ def edit(self, user, session=None, **kwargs):
                 setattr(self, name, value)
         session.add(self)
 
+    @SessionManager.with_session
+    def archive_project(self, user, session=None):
+        """"Archives the project"""
```
Comment (caydenreynolds) [inline src/ypd/model/project.py:110]: This function doesn't archive the project. This functions toggles the archived flag. The name is confusing, and the docstring is inaccurate

Comment (gabeholley222) [reply src/ypd/model/project.py]: Changed function to be toggle_archived and comment to reflect. Made ability to restore project to Archived = False only for admins.

File: src/ypd/servlets/project_servlet.py (hunk 2 above)
Comment (gabeholley222) [reply src/ypd/servlets/project_servlet.py]: Removed methods as they weren't needed.

File: src/ypd/templates/project.html (hunk 1 above)
Comment (gabeholley222) [reply src/ypd/templates/project.html]: Added two icons, one to archive flag true and one to archive flag false.

File: src/ypd/servlets/project_servlet.py [hunk 4]
```diff
@@ -77,7 +88,7 @@ def submit(self):
                 else:
                     project = Solicited()
                 project.post(title, description, current_user)
-                return redirect(url_for('IndexView:get',
+                return redirect(url_for('IndexView:view',
```
Comment (caydenreynolds) [inline src/ypd/servlets/project_servlet.py:91]: This should be 'IndexView:get' as per my bugfix the other day

File: src/ypd/templates/project.html [hunk 5]
```diff
@@ -28,6 +28,20 @@ <h1>{{project.title}}</h1>
...
+                    {% if project.archived == false %}
+                        🗑️
+                    {% else %}
+                        🔓 
+                    {% endif %}
+                </button>
+            </a>
+        </div>
+    {% endif %}
+
```
Comment (caydenreynolds) [inline src/ypd/templates/project.html:44]: You can only see this if you're a poster or an admin. 
What if an unrelated user looks at the project? Shouldn't they know it's been archived?

File: src/ypd/servlets/project_servlet.py (hunk 4 above)
Comment (gabeholley222) [reply src/ypd/servlets/project_servlet.py]: Fixed it, not sure why it changed it as i never played with your code other than to copy paste.

File: src/ypd/templates/project.html (hunk 5 above)
Comment (gabeholley222) [reply src/ypd/templates/project.html:53]: Added an additional button(display only) that displays a lock icon if archive = true for all users to see.

**Comment by gabeholley222** [comment]
Added tooltips with meaningful messages

**Comment by gabeholley222** [comment]
Committed resolution so that only admin can see unlock button.

**Pull request merged**